### PR TITLE
Update vision_encoder.cpp for miniCPM

### PR DESCRIPTION
**Pull request by wenyi5608**
Modified the visual image functions that convert NCHW to NCkernel (HW/kernel*kernel) to match the Python implementation

### CODE REVIEW

File: src/cpp/src/visual_language/vision_encoder.cpp [hunk 1]
```diff
@@ -309,35 +309,76 @@ EncodedImage llava_image_embed_make_with_bytes_slice(clip_ctx& ctx_clip, const o
...
+    
+    //image chw to 1*c*kernel*hw/kernel*kernel and padding zero
     const clip_image_f32& resized_preprocessed = preprocessed.at(0).at(0);
-    std::copy(resized_preprocessed.buf.begin(), resized_preprocessed.buf.end(), batched_data);
+    size_t img_h = resized_preprocessed.ny;
+    size_t img_w = resized_preprocessed.nx;
+    ov::Tensor clip_img{ov::element::f32, {1, channels, img_h, img_w}};
+    float* clip_data = clip_img.data<float>();
+    std::copy(resized_preprocessed.buf.begin(), resized_preprocessed.buf.end(), clip_data);
+    ov::Tensor clip_pixel_values = preprocess_for_encoder(clip_img, patch_size);
```
Comment (Wovchena) [inline src/cpp/src/visual_language/vision_encoder.cpp:342]:
```suggestion
    ov::Tensor clip_img{ov::element::f32, {1, channels, img_h, img_w}, resized_preprocessed.buf.data()};
    ov::Tensor clip_pixel_values = preprocess_for_encoder(clip_img, patch_size);
```
Can you try this. Same for `clip_img` below.

File: src/cpp/src/visual_language/vision_encoder.cpp [hunk 2]
```diff
@@ -309,35 +309,76 @@ EncodedImage llava_image_embed_make_with_bytes_slice(clip_ctx& ctx_clip, const o
...
         return processed_row;
     });
 
-    ov::Tensor batched_images{ov::element::f32, {n_images, 3, max_h, max_w}};
-    float* batched_data = batched_images.data<float>();
+    ov::Tensor pixel_values{ov::element::f32, {n_images, channels, patch_size, max_size / patch_size}};
+    size_t d3_all_pixel = pixel_values.get_shape().at(3);
+    float* pixel_value_data = pixel_values.data<float>();
+    
+    //image chw to 1*c*kernel*hw/kernel*kernel and padding zero
```
Comment (Wovchena) [inline src/cpp/src/visual_language/vision_encoder.cpp:335]:
```suggestion
    //image chw to 1*c*kernel*hw/kernel and padding zero
```

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: src/cpp/src/visual_language/vision_encoder.cpp (hunk 1 above)
Comment (wenyi5608) [reply src/cpp/src/visual_language/vision_encoder.cpp]: According to your suggestion, I updated the code. Thanks!

**Review by Wovchena — APPROVED** [review]

**Comment by Wovchena** [comment]
build_jenkins

**Pull request merged**